Objects/complexobject.c

```diff
@@ -966,7 +925,9 @@ complex_new_impl(PyTypeObject *type, PyObject *r, PyObject *i)
     }
 
     nbr = Py_TYPE(r)->tp_as_number;
-    if (nbr == NULL || (nbr->nb_float == NULL && nbr->nb_index == NULL)) {
+    if (nbr == NULL ||
+        (nbr->nb_float == NULL && nbr->nb_index == NULL && !PyComplex_Check(r)))
+    {
         PyErr_Format(PyExc_TypeError,
                      "complex() first argument must be a string or a number, "
```

Comment:
Since you change the float error, would it make sense to write "a complex number" here?

Reply:
It can be a real number (int, float, Fraction, Decimal, etc). I think that requiring the first argument of the complex constructor to be a complex number would be confusing.

Reply:
An "integer" number is also a real number, and a real number is also a complex number, no? I'm thinking at "real" and "complex" such as in the numbers tower. But it's just a remark, you can leave it as it is.

Reply:
In mathematics it is true. But `int` and `float` are not subclasses of `complex` in Python.

Also, requiring an argument of the `complex` constructor to be a `complex` number would be very confusing.

Reply:
> In mathematics it is true. But int and float are not subclasses of complex in Python.

Oh, I didn't look at that and you're right :-)